### ISSUE TITLE
Auto-update checkpoints file to avoid replay/ repair degradation 

### ISSUE THREAD

**Issue by jim618** [bitcoin-solutions/multibit-hd#720]
Currently the checkpoints file is updated only when there is a new version of MBHD.
Thus as it gets older replays/ repairs get slower as we always do a replay from a checkpoint.

Checkpoints occur at difficulty changes, about every 2000 blocks / fortnight.

It would be possible for MBHD (and Classic) to update the checkpoints file themselves. If you have a look at BuildCheckpoints the code isn't terribly complicated.

This would mean that the performance of replays/ repairs for MBHD would not degrade over time when there were infrequent/ no MBHD releases (or Classic releases if the code were backported).

**Comment by gary-rowe** [comment]
Agreed. We discussed this a long time back for MBHD but decided it wasn't on the critical path to MVP. We now have the opportunity to do it. 

Adding to 0.1.5.
